app/controllers/api/v1/animals_controller.rb

```diff
@@ -1,35 +1,42 @@
+require 'active_support/inflector'
+
 class Api::V1::AnimalsController < ApplicationController
 
+  def index
+    animal = Animal.find_by(id: param[:id])
+    if params[:action_type] == "eat_me"
+
+      animal_predators = rabbit.predators.split(',').map(&:strip).first(3)
+        predators_data = predators.map do |predator_name|
+      # render json: AnimalSerializer.new(predators_data)
+        end
+    end
+  end
+
   def create
-    if params[:action_type] == "start"
-      rabbit = Animal.find_by(name: "rabbit")
+    if params[:action_type] == "start" || params[:action_type] == "selected_animal"
```

Comment:
I love that we only had to change this a little, adding animal or rabbit made this so much more dynamic and usable and dry.